File: sources/api/Models/Debt.cs
namespace DotNetAPI.Model
{
    public class Debt
    {
        public int Id { get; set; }
        public required int GroupId { get; set; }
        public required Group Group { get; set; }
        public required int BillId { get; set; }
        public required Expense Expense{ get; set; }
        public required int UserIdInCredit { get; set; }
        public required User UserInCredit { get; set; }
        public required int UserIdInDebt { get; set; }
        public required User UserInDebt { get; set; }
        public required float Amount { get; set; }
        public required bool IsPaid { get; set; }
        public required bool IsCanceled { get; set; }
    }
}

File: sources/mobile/app/src/main/java/com/console/ratcord/domain/entity/debt/Debt.kt
package com.console.ratcord.domain.entity.debt

import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.PrimaryKey
import com.console.ratcord.domain.entity.category.Category
import com.console.ratcord.domain.entity.expense.Expense
import com.console.ratcord.domain.entity.group.Group
import com.console.ratcord.domain.entity.user.User
import com.console.ratcord.domain.entity.user.UserExtraMinimal
import com.console.ratcord.domain.entity.user.UserMinimal
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
@Entity
data class Debt(
    @PrimaryKey val id: Int,
    val userInCredit: UserExtraMinimal,
    val userInDebt: UserExtraMinimal,
    val amount: Float,
    val isPaid: Boolean,
    val isCanceled: Boolean
    )